### DISCUSSION TITLE
Loading specific state from path or id

### DISCUSSION THREAD

**Discussion by hiroMTB** [statelyai/xstate#2696] [Q&A] [answered]
Hello, I'm trying to load a specific state from a known path such as #aaa.bbb.ccc

I found 2 methods, [getStateNodeById()](https://xstate.js.org/api/interfaces/statemachine.html#getstatenodebyid) and [getStateNodeByPath()](https://xstate.js.org/api/interfaces/statemachine.html#getstatenodebypath) in StateMachine class.
But both the functions return StateNode, which I can't pass to [service.start()](https://xstate.js.org/api/classes/interpreter.html#start), which needs State or StateValue.
How could I achieve this?

Here is my current code.

```
const path = "#aaa.bbb.ccc"
const stateNode = machine.getStateNodeByPath(path)

// const id = "aaa.bbb.ccc"
// const stateNode = machine.getStateNodeById(id)

// somehow I have to convert stateNode to State (or StateValue)
const state = ???

service.stop()
service.start(state)

```

**Comment by mattpocock** [comment, marked as answer]
`State.from('id')`

**Reply by hiroMTB** [reply]
Wow, Thanks for your super quick answer! Will try it now.

**Reply by hiroMTB** [reply]
Ok, so I noticed `State.from('id')` does not work in my case.
I can move to a specific state, but then my service(interpreter) stops after transition.
My machine config is like below.
```
export const myMachine =
{
  id: 'myId',
  initial: 'title',
  states: {
    title: {
      after: {
        1000: {
          target: "somewhere",
        }
      }
    },
...
```

And I found this issue https://github.com/statelyai/xstate/issues/2434
Using [getInitialState](https://xstate.js.org/api/interfaces/statemachine.html#getinitialstate) instead solved the problem.
Thanks anyway!

**Reply by mattpocock** [reply]
Nice, good stuff!

**Reply by imran-baitham** [reply]
Nice
